File: gdscript/src/main/java/gdscript/psi/GdClassVarDeclTl.java
package gdscript.psi;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.StubBasedPsiElement;
import gdscript.index.stub.GdClassVarDeclStub;
import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface GdClassVarDeclTl extends GdTopLevelDecl, StubBasedPsiElement<GdClassVarDeclStub>, GdDocumented {

  @Nullable
  GdAssignTyped getAssignTyped();

  @Nullable
  GdEndStmt getEndStmt();

  @Nullable
  GdExpr getExpr();

  @Nullable
  GdSetgetDecl getSetgetDecl();

  @Nullable
  GdTyped getTyped();

  @Nullable
  GdVarNmi getVarNmi();

  @NotNull
  String getName();

  @NotNull
  Boolean isStatic();

  @NotNull
  String getReturnType();

  @NotNull
  ItemPresentation getPresentation();

  boolean isAnnotated(@NotNull String annotator);

}

File: gdscript/src/main/java/gdscript/psi/GdClassVarDeclTl.kt
package gdscript.psi

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.StubBasedPsiElement
import gdscript.index.stub.GdClassVarDeclStub
import gdscript.psi.types.GdDocumented

interface GdClassVarDeclTl : GdTopLevelDecl, StubBasedPsiElement<GdClassVarDeclStub>, GdDocumented {
    val assignTyped: GdAssignTyped?

    val endStmt: GdEndStmt?

    val expr: GdExpr?

    val setgetDecl: GdSetgetDecl?

    val typed: GdTyped?

    val varNmi: GdVarNmi?

    fun getName(): String

    val isStatic: Boolean

    val returnType: String

    fun getPresentation(): ItemPresentation

    fun isAnnotated(annotator: String): Boolean
}
